### PR TITLE
test PRs branches

### PR DESCRIPTION
**Pull request by veronikatimkina**

### CODE REVIEW

**Comment by testaccount120** [comment]
уруеруерер

**Comment by veronikatimkina** [comment]
еекр

**Comment by veronikatimkina** [comment]
аврверуеруер

**Comment by veronikatimkina** [comment]
пкпукпукпкуп

**Comment by veronikatimkina** [comment]
рукрукккур

**Comment by veronikatimkina** [comment]
еркркерек

**Comment by veronikatimkina** [comment]
варварварвар

File: src/TestBr.java [hunk 1]
```diff
@@ -0,0 +1,2 @@
+public class TestBr {
```
Comment (veronikatimkina) [inline src/TestBr.java:1]: > test

lalala

File: src/TestBr.java [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+}
```
Comment (veronikatimkina) [inline src/TestBr.java:2]: test
ddd

**Comment by veronikatimkina** [comment]
efwefewfewf
efef

**Review by veronikatimkina — COMMENTED** [review]
> test quotes

no quotes

**Comment by veronikatimkina** [comment]
> Are classes also expected to force object lifetime rules or is it upto the user (trusting he knows what he's doing).

Don't fully understand this question. Can you rephrase it?

But I still think this case should be an exception. This is a kind of exception that should never happen in real life, as we should make sure that we properly destruct everything in order during shutdown. For example when the user calls device.disconnect(). During disconnect I would expect the logging "thread" to stop waiting for events, keepalives to stop etc, before dispatcher & endpoint is destructed and the camera finally disconnected (SampleTestClass destructed). Nothing should "hang" waiting for data after the disconnecter has been killed, that suggests a problem in the design. So the exception is a way to detect this kind of issue/bug as soon as possible.

Of course user can get into problem by doing device.capture() in one thread and device.disconnect() in another thread at the same time. But we are already saying that the device object is not thread safe. So we can assume that disconnect is not done while capturing.

File: src/TestBr.java (hunk 2 above)
Comment (veronikatimkina) [reply src/TestBr.java:2]: Are classes also expected to force object lifetime rules or is it upto the user (trusting he knows what he's doing).
> Don't fully understand this question. Can you rephrase it?
But I still think this case should be an exception. This is a kind of exception that should never happen in real life, as we should make sure that we properly destruct everything in order during shutdown. For example when the user calls device.disconnect(). During disconnect I would expect the logging "thread" to stop waiting for events, keepalives to stop etc, before dispatcher & endpoint is destructed and the camera finally disconnected (SampleTestClass destructed). Nothing should "hang" waiting for data after the disconnecter has been killed, that suggests a problem in the design. So the exception is a way to detect this kind of issue/bug as soon as possible.
Of course user can get into problem by doing device.capture() in one thread and device.disconnect() in another thread at the same time. But we are already saying that the device object is not thread safe. So we can assume that disconnect is not done while capturing.

File: src/TestBr.java (hunk 1 above)
Comment (veronikatimkina) [reply src/TestBr.java:1]: > **test bold quote**

no quote

Comment (veronikatimkina) [reply src/TestBr.java:1]: first empty line


two empty



three

Comment (veronikatimkina) [reply src/TestBr.java:1]: okay


2

Comment (veronikatimkina) [inline src/TestBr.java:1]: s
i
n
g
l
e


comm

**Review by veronikatimkina — COMMENTED** [review]
submit


review 



with paragraph

File: src/TestBr.java (hunk 1 above)
Comment (veronikatimkina) [reply src/TestBr.java:1]: > `public` alala
